registry/remote/repository.go

```diff
@@ -739,6 +796,58 @@ func (s *manifestStore) Resolve(ctx context.Context, reference string) (ocispec.
 	}, nil
 }
 
+// FetchTag fetches the content identified by a reference.
+// It is equivalent to call `Resolve()` and then `Fetch()`
```

Comment:
This should not be equivalent of resolve and fetch. They are 2 API requests where are this should be only one GET request.

Reply:
It is equivalent in terms of functionality. We need to mention that it is more efficient as it has only 1 GET request in the documentation.

Reply:
I would agree but would like to keep the docs for the function without ambiguity. It is clearly 2 APIs and they have different behaviors depending on intermediaries.  

Maybe edit to keep it concise and have more details in the example. 
`//FetchTag fetches the content identified by the reference. The reference may be a tag or digest.`

Reply:
On now thinking more - its sounds incorrect to call this method `FetchTag` if we are passing a digest. 
We should consider some other name like `FetchByReference(string reference....)`. 
Or am I missing something here?

Reply:
Agreed. I've rename it to `FetchReference` and made the doc concise.